### PR TITLE
[tests-only][full-ci]Add /Shares related tests for lock property timeout

### PR DESCRIPTION
**Pull request by SagarGi**
### Description

This PR adds the tests for lock related to /Shares for `timout` properties which is removed from this PR https://github.com/owncloud/core/pull/40300 as `/Shares` related implementation is not in core and is shifted to ocis.

### Related issue
https://github.com/owncloud/ocis/issues/4154#issuecomment-1206026045

Note: Most of the tests for the `etag` check when deleting files and folder is run from core API suites on ocis. This test added on ocis is only related to `/Shares` (Shares Jail)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -3016,6 +3101,115 @@ public function userHasStoredEtagOfElementOnPathFromSpace($user, $path, $storePa
...
+     * @return void
+     * @throws Exception
+     */
+    public function userGetsPropertiesOfFolderOfSpaceAndAssertValueOfItemInResponseToUserRegExp(string $user, string $xpath, string $path, string $spaceName, string $pattern):void {
+        $space = $this->getSpaceByName($user, $spaceName);
+        $fullUrl = $space['root']['webDavUrl'] . '/' . ltrim($path, '/');
+        $propertiesTable = new TableNode([['propertyName'],['d:lockdiscovery']]);
+        $properties = null;
+        $this->featureContext->verifyTableNodeColumns($propertiesTable, ["propertyName"]);
+        $this->featureContext->verifyTableNodeColumnsCount($propertiesTable, 1);
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3176]: the table is self-created, so validations are of no use they will always pass
also, a table would be better for these long arguments IMO.
that would also help to check for multiple resources.

File: tests/acceptance/features/apiSpaces/lockSpaces.feature [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    And user "Brian" uploads a file "filesForUpload/textfile.txt" inside space "Shares Jail" to "PARENT/textfile.txt" using the WebDAV API
+    And user "Alice" uploads file "filesForUpload/textfile.txt" to "/PARENT/textfile.txt" using the WebDAV API
+    Then the HTTP status code of responses on all endpoints should be "423"
+    And as "Alice" file "/PARENT/textfile.txt" should not exist
+    Examples:
+      | lock-scope |
+      | shared     |
+      | exclusive  |
+
+  Scenario Outline: set timeouts of LOCKS on shares as owner set timeout on folder as receiver check it
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/lockSpaces.feature:95]: extra line for tags

File: tests/acceptance/features/apiSpaces/lockSpaces.feature [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    And as user "Brian" the lock discovery property "//d:timeout" of the folder "/PARENT/parent.txt" inside space "Shares Jail" should match "<result>"
+    Examples:
+      | timeout         | result          |
+      | second-999      | /Second-\d{3}$/ |
+      | second-99999999 | /Second-\d{5}$/ |
+      | infinite        | /Second-\d{5}$/ |
+      | second--1       | /Second-\d{5}$/ |
+      | second-0        | /Second-\d{4}$/ |
+
+  Scenario Outline: set timeouts of LOCKS on shares as share receiver set timeout on folder as owner check it
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/lockSpaces.feature:117]: extra line for tags

**Comment by SagarGi** [comment]
This tests is folder lock related and can be closed with this comment https://github.com/owncloud/ocis/issues/1284#issuecomment-1238336767 and this issue https://github.com/owncloud/ocis/issues/4526

**Pull request closed without merging**